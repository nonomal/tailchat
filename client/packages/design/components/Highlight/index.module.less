.highLight {
  background-color: rgba(0,0,0,0.2);
  border-radius: 0.25rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}
