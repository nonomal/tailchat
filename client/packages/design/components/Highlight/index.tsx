import React, { PropsWithChildren } from 'react';
import styles from './index.module.less';

export const Highlight: React.FC<PropsWithChildren> = React.memo((props) => {
  return <span className={styles.highLight}>{props.children}</span>;
});
Highlight.displayName = 'Highlight';
