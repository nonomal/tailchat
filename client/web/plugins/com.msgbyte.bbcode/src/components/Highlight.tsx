import React from 'react';
import CodeRender from 'react-highlight';

import 'highlight.js/styles/default.css';

const Highlight: React.FC<{
  language: string;
  code: string;
}> = React.memo((props) => {
  const CodeComponent = (CodeRender as any).default as typeof CodeRender; // TODO: ministar编译问题，先跳过
  return <CodeComponent className={props.language}>{props.code}</CodeComponent>;
});
Highlight.displayName = 'Highlight';

export default Highlight;
