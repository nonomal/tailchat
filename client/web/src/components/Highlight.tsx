export { Highlight } from 'tailchat-design';
