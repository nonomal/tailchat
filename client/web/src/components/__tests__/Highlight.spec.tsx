import { render } from '@testing-library/react';
import React from 'react';
import { Highlight } from '../Highlight';

describe('Highlight', () => {
  test('render', () => {
    const wrapper = render(<Highlight>Any Text</Highlight>);
    expect(wrapper.container).toMatchSnapshot();
  });
});
