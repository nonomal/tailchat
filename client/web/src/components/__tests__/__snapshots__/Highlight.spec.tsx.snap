// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight render 1`] = `
<div>
  <span
    class="highLight"
  >
    Any Text
  </span>
</div>
`;
